exit: Ignore ExitError error type

jcvenegas opened the pull request:
When the agent waits for a process, if the process
terminates with an exit code different to 0, the
function will return an error of type ExitError.
The process exit code is not relevant for the
agent, lets ignore that error type.

Fixes: #146

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

sboeuf commented:
@jcvenegas I think you should fix the commit message "when the agent wait for the agent", and fix the comment in the code. Other than that, I like the patch!
LGTM

[![Approved with PullApprove](https://img.shields.io/badge/approvers-approved-brightgreen.svg)](https://pullapprove.com/clearcontainers/agent/pull-request/147/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent) [![Approved with PullApprove](https://img.shields.io/badge/reviewers-approved-brightgreen.svg)](https://pullapprove.com/clearcontainers/agent/pull-request/147/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent)

sboeuf commented on agent.go:
```diff
@@ -854,10 +855,10 @@ func (p *pod) runContainerProcess(cid, pid string, terminal bool, started chan e
 
 	processState, err := proc.process.Wait()
-	if err != nil {
+	// Ignore error if process fail because of an unsuccessful exit code
```
s/fail/fails/

jcvenegas committed:
exit: Ignore ExitError error type

When the agent waits for a process, if the process
terminates with an exit code different to 0, the
function will return an error of type ExitError.
The process exit code is not relevant for the
agent, lets ignore that error type.

Fixes: #146

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

jcvenegas commented:
@sboeuf changed commit message. Fixed typo, ready to merge :) .
